src/app/services/dia-backend/asset/dia-backend-asset-repository.service.ts

```diff
@@ -1,87 +1,54 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { MatSnackBar } from '@angular/material/snack-bar';
 import { TranslocoService } from '@ngneat/transloco';
-import { isEqual } from 'lodash';
+import { BehaviorSubject, defer, forkJoin, merge } from 'rxjs';
 import {
-  BehaviorSubject,
-  defer,
-  EMPTY,
-  forkJoin,
-  from,
-  iif,
-  merge,
-  Observable,
-  of,
-} from 'rxjs';
-import {
-  catchError,
   concatMap,
   concatMapTo,
-  delay,
-  distinctUntilChanged,
-  first,
   map,
-  mapTo,
   pluck,
   single,
   tap,
-  toArray,
 } from 'rxjs/operators';
 import { base64ToBlob } from '../../../utils/encoding/encoding';
 import { toExtension } from '../../../utils/mime-type';
-import { switchTap, VOID$ } from '../../../utils/rx-operators/rx-operators';
-import { Database } from '../../database/database.service';
-import { OnConflictStrategy, Tuple } from '../../database/table/table';
-import { ImageStore } from '../../image-store/image-store.service';
+import { Tuple } from '../../database/table/table';
 import { NotificationService } from '../../notification/notification.service';
 import {
-  getOldProof,
   getOldSignatures,
-  getProof,
   getSortedProofInformation,
   OldSignature,
   SortedProofInformation,
 } from '../../repositories/proof/old-proof-adapter';
 import { Proof } from '../../repositories/proof/proof';
-import { ProofRepository } from '../../repositories/proof/proof-repository.service';
 import { DiaBackendAuthService } from '../auth/dia-backend-auth.service';
 import { BASE_URL } from '../secret';
 
 @Injectable({
   providedIn: 'root',
 })
 export class DiaBackendAssetRepository {
-  private readonly table = this.database.getTable<DiaBackendAsset>(
-    DiaBackendAssetRepository.name
-  );
   private readonly _isFetching$ = new BehaviorSubject(false);
-  private isDirty = true;
+
+  private readonly fetchAllCache$ = new BehaviorSubject<DiaBackendAsset[]>([]);
```

Comment:
By using a in-memory cache, does that mean it's not possible to determine if an asset has been uploaded or not if the user is offline?

For example, will the user see the `!` mark on all previous captures when trying to use Capture offline?